subworkflows/local/prepare_ncbi_taxinfo.nf

```diff
@@ -0,0 +1,136 @@
+        """
+}
+
+process DOWNLOAD_ACESSION2TAXID {
+    tag "Downloading prot.accession2taxid.gz"
+    label 'process_low'
+    storeDir "${params.prot_accession2taxid_gz_dir}"
+
+    conda (params.enable_conda ? "conda-forge::rsync=3.2.3" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/YOUR-TOOL-HERE"
+    } else {
+        container "jason-c-kwan/autometa:${params.autometa_image_tag}"
+    }
+
+    output:
+        // hack nf-core options.args3 and use for output name
+        path "prot.accession2taxid.gz" , emit: accession2taxid
+        path  "*.version.txt"   , emit: version
+    script:
+        """
+        rsync -a \\
+            --quiet \\
+            'rsync://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/prot.accession2taxid.gz' 'prot.accession2taxid.gz'
+
+        rsync -a \\
+            --quiet \\
+            'rsync://ftp.ncbi.nlm.nih.gov/pub/taxonomy/accession2taxid/prot.accession2taxid.gz.md5' 'prot.accession2taxid.gz.md5'
+
+        md5sum -c *.md5
+
+        rsync --version | head -n1 > rsync.version.txt
+        """
+}
+
+
+process DOWNLOAD_TAXDUMP {
+    tag "Downloading taxdump.tar.gz"
+    label 'process_low'
+    storeDir "${params.taxdump_tar_gz_dir}"
+
+    conda (params.enable_conda ? "conda-forge::rsync=3.2.3" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/YOUR-TOOL-HERE"
+    } else {
+        container "jason-c-kwan/autometa:${params.autometa_image_tag}"
+    }
+
+    output:
+        path "*" , emit: taxdump_files
+        path  "*.version.txt"   , emit: version
+
+    script:
+        """
+        rsync -a \
+            --quiet \
+            'rsync://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz' 'taxdump.tar.gz'
+
+        rsync -a \
+            --quiet \
+            'rsync://ftp.ncbi.nlm.nih.gov/pub/taxonomy/taxdump.tar.gz.md5' 'taxdump.tar.gz.md5'
+
+        md5sum -c *.md5
+        rm 'taxdump.tar.gz.md5'
+        tar -xf taxdump.tar.gz
+        rm taxdump.tar.gz
+
+        rsync --version | head -n1 > rsync.version.txt
+        """
+}
+
+
+workflow PREPARE_TAXONOMY_DATABASES {
+    main:
+        taxdump_dir = file(params.taxdump_tar_gz_dir)
+        taxdump_dir_files = taxdump_dir.list()
+        expected_files = ['citations.dmp', 'delnodes.dmp', 'division.dmp', 'gencode.dmp', 'merged.dmp', 'names.dmp', 'nodes.dmp']
+
+        if (taxdump_dir_files.containsAll(expected_files)){
+            taxdump_files = taxdump_dir_files
+        } else {
+            DOWNLOAD_TAXDUMP()
+            DOWNLOAD_TAXDUMP.out.taxdump_files
+                .set{taxdump_files}
+        }
+
+        accession2taxid_dir = file(params.prot_accession2taxid_gz_dir)
+        accession2taxid_dir_files = accession2taxid_dir_files.list()
```

Comment:
@WiscEvan just a note- you had a  commit that removed `.list()` but kept `.collect()`.
That creates a list that's basically each element of the path. Keeping `.list()` creates the filenames of the files in the provided directory, which is what was intended. 
You were right that both was unnecessary